src/ray/object_manager/object_manager.cc

```diff
@@ -24,17 +24,19 @@ ObjectManager::ObjectManager(asio::io_service &main_service,
     : client_id_(gcs_client->client_table().GetLocalClientId()),
       config_(config),
       object_directory_(new ObjectDirectory(gcs_client)),
-      store_notification_(main_service, config_.store_socket_name),
+      main_service_(main_service),
+      store_notification_(main_service_, config_.store_socket_name),
       // release_delay of 2 * config_.max_sends is to ensure the pool does not release
       // an object prematurely whenever we reach the maximum number of sends.
       buffer_pool_(config_.store_socket_name, config_.object_chunk_size,
-                   /*release_delay=*/2 * config_.max_sends),
+                   /* release_delay */ 0),
```

Comment:
If the object manager is transferring large objects, we don't want it to hold on to them and lock up memory in the object store.